### PR TITLE
feat(server): workspace settings infrastructure and usecase

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
- This PR implements the workspace settings infrastructure and usecase layers

### CODE REVIEW

File: server/schemas/workspace.graphql [hunk 1]
```diff
@@ -82,6 +87,7 @@ input UpdateIntegrationOfWorkspaceInput {
 
 input DeleteWorkspaceInput {
     workspaceId: ID!
+    settingsId: ID!
```
Comment (yk-eukarya) [inline server/schemas/workspace.graphql:90]:
```suggestion
```

File: server/schemas/workspace.graphql [hunk 2]
```diff
@@ -40,6 +41,10 @@ input CreateWorkspaceInput {
 input UpdateWorkspaceInput {
     workspaceId: ID!
     name: String!
+    settingsId: ID!
+    avatar: String
+	tiles: WorkspaceResourceListInput
+	terrains: WorkspaceResourceListInput
```
Comment (yk-eukarya) [inline server/schemas/workspace.graphql:47]:
```suggestion
    avatar: String
    tiles: WorkspaceResourceListInput
    terrains: WorkspaceResourceListInput
```

File: server/schemas/workspacesettings.graphql
Comment (yk-eukarya) [inline server/schemas/workspacesettings.graphql:1]: be careful about line indentation

File: server/schemas/workspace.graphql (hunk 1 above)
Comment (nourbalaha) [reply server/schemas/workspace.graphql]: Done!

File: server/schemas/workspace.graphql (hunk 2 above)
Comment (nourbalaha) [reply server/schemas/workspace.graphql]: Done!

File: server/schemas/workspacesettings.graphql
Comment (nourbalaha) [reply server/schemas/workspacesettings.graphql:1]: Okay.

File: server/schemas/workspacesettings.graphql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+type ResourceList {
+    resources: [Resource!]!
+    selectedResource: ID
+    enabled: Boolean # only in terrains
+}
+
+type Resource {
+	id: ID!
+    type: String!
```
Comment (yk-eukarya) [inline server/schemas/workspacesettings.graphql:15]: use an enum for type

File: server/schemas/workspacesettings.graphql [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    name: String!
+    url: String!
+    image: String!
+    cesiumIonAssetId: String # only in terrains
+    cesiumIonAccessToken: String # only in terrains
+}
```
Comment (yk-eukarya) [inline server/schemas/workspacesettings.graphql:21]:
```suggestion
type Resource {
	id: ID!
    type: ResourceType!
    props: ResorceProps
}

type ResorceProps @onlyOne{
    url: UrlResorceProps
    cesium: CesiumResorceProps
}

type UrlResorceProps {
    name: String!
    url: String!
    image: String!
}

type CesiumResorceProps {
    name: String!
    url: String!
    image: String!
    cesiumIonAssetId: String! 
    cesiumIonAccessToken: String!
}
```

**Review by yk-eukarya — APPROVED** [review]

**Pull request merged**